### ISSUE TITLE
Publish datasource and connection elements

### ISSUE THREAD

**Issue by jorwoods** [tableau/server-client-python#1057]
**Describe the bug**
It appears to me that the publish datasource method utilizes a `connectionCredentials` subelement, and not a `connections` element. 

https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_data_sources.htm#publish_data_source

The way I see it, we could solve this by one of two routes:

1. Combine the connections and connection_credentials inside the function call to a single object that gets passed to the publish request factory.
2. Make connection credentials accept a sequence, and surface an error to TSC users when they provide both.

**Versions**
Details of your environment, including:
 - Tableau Server version (or note if using Tableau Online) N/A
 - Python version 3.8.10
 - TSC library version 0.19.0

**Comment by jacalata** [comment]
Yes - I'm trying to figure out how this happened, but so far it's a mystery. 
I think we want to end up with just the connection_credentials option as a sequence, but I'll take a look to see if there's a way we can temporarily support both inputs.

**Comment by jorwoods** [comment]
Circling back to this. I think the best option might be to mirror the `workbook.publish` endpoint and have it be `connections=Sequence[ConnectionItem]`. I think that would support the most scenarios and have the benefit of keeping the workbook and datasource endpoints as similar.

**Comment by jacalata** [comment]
The datasource publish signature now mirrors workbook.publish: `datasources.publish(..., connection_credentials=None, connections=None, ...)` accepts either a single `ConnectionCredentials` OR a `Sequence[ConnectionItem]`, and passing both raises `RuntimeError('You cannot set both connections and connection_credentials')` (see `tableauserverclient/server/request_factory.py` `DatasourceRequest._generate_xml`). This matches the recommendation in the last comment on this thread. Closing as completed and removing the `in-progress` label.
